app/controllers/application_controller.rb

```diff
@@ -1,2 +1,32 @@
 class ApplicationController < ActionController::Base
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate_request
+
+  private
+
+  def authenticate_request
+    header = request.headers['Authorization']
+    token = header&.split&.last
+
+    return render json: { error: 'Unauthorized' }, status: :unauthorized unless token
+
+    begin
+      decoded_token = decode_token(token)
+    rescue JWT::ExpiredSignature
+      return render json: { error: 'Token has expired' }, status: :unauthorized
```

Comment:
Now that you have introduced I18n localization, can we move these hardcoded errors to the en.yaml file?

Reply:
This applies to all such use-cases